src/MgvOfferTaking.sol

```diff
@@ -306,146 +306,97 @@ abstract contract MgvOfferTaking is MgvHasOffers {
     }
   }
 
-  /* # Sniping */
-  /* ## Snipes */
-  //+clear+
-
-  /* `snipes` executes multiple offers. It takes a `uint[4][]` as penultimate argument, with each array element of the form `[offerId,tick,fillVolume,offerGasreq]`. The return parameters are of the form `(successes,snipesGot,snipesGave,bounty,feePaid)`. 
-  Note that we do not distinguish further between mismatched arguments/offer fields on the one hand, and an execution failure on the other. Still, a failed offer has to pay a penalty, and ultimately transaction logs explicitly mention execution failures (see `MgvLib.sol`). */
-
-  function snipes(address outbound_tkn, address inbound_tkn, uint[4][] calldata targets, bool fillWants)
-    external
-    returns (uint, uint, uint, uint, uint)
-  {
-    unchecked {
-      return generalSnipes(outbound_tkn, inbound_tkn, targets, fillWants, msg.sender);
-    }
-  }
-
-  /*
-     From an array of _n_ `[offerId, tick,fillVolume,gasreq]` elements, execute each snipe in sequence. Returns `(successes, takerGot, takerGave, bounty, feePaid)`. 
-
-     Note that if this function is not internal, anyone can make anyone use Mangrove.
-     Note that unlike general market order, the returned total values are _not_ `mor.totalGot` and `mor.totalGave`, since those are reset at every iteration of the `targets` array. Instead, accumulators `snipesGot` and `snipesGave` are used. */
-  function generalSnipes(
+  /* # Cleaning */
+  // FIXME: Document cleaning
+  // FIXME: Maybe `cleanFailingOffer` instead?
+  /* Any `taker` can be impersonated when cleaning because the function reverts if the offer succeeds, cancelling any token transfers. And after a `clean` where the offer has failed, all token transfers have been reverted -- but the sender will still have received the bounty of the failing offers. */
+  // FIXME: This would be simpler if tick had to match exactly: Then `fillWants` wouldn't be needed and `fillVolume` could be simplified to either `takerWants` or `takerGives`.
```

Comment:
Do we have a good use case for non-exact match? Simplified code seems like a win.

Reply:
I'm wondering about the opposite: should we completely remove the price match? Initially it's there for "normal snipes" (to protect the taker). In the absence of normal snipes, what happens if we just ignore the price?

Reply:
I think the cleaner risks spending more gas on failing to make the offer fail: The offer is now more expensive and no longer fails.

Reply:
The scenario is that the maker updates the offer between us making the decision to clean and actually cleaning - I don't think we can a priori say anything about whether it would still fail. Skipping early saves gas - in case of failure still, we win gas. I don't know what is most common.
I lean towards skipping since information has changed - spending gas on executing the entire offer if it succeeds is quite wasteful.